reinforcement_learning/examples/override_interface/override_interface.cc

```diff
@@ -0,0 +1,109 @@
+/**
+ * @brief model inference API example showing custom overriden interface
+ *
+ * @file override_interface.cc
+ * @author Jack Gerrits et al
+ * @date 2018-08-20
+ */
+#include <iostream>
+#include <fstream>
+
+#include "config_utility.h"
+#include "live_model.h"
+#include "factory_resolver.h"
+#include "constants.h"
+#include "logger.h"
+
+// Namespace manipulation for brevity
+namespace r = reinforcement_learning;
+namespace u = r::utility;
+namespace err = r::error_code;
+
+// Custom implementations must inherit from the respective i_* abstract class.
+class ostream_logger : public r::i_logger {
+public:
+  ostream_logger(std::ostream& stream)
+    : _stream(stream)
+  {}
+
+  virtual int init(r::api_status* status) override {
+    return err::success;
+  }
+
+protected:
+  virtual int v_append(std::string& data, r::api_status* status) override {
```

Comment:
>v_append [](start = 14, length = 8)

Possible thread safety:  observation and interaction threads will access 2 different copies of ostream_logger.  However since we pass same instance of ostream to both loggers, won't they write to the same underlying std::ostream object?

Reply:
Agreed, for the example we can pass a mutex in for it to use.